src/ConstructiveSolidGeometry/DistanceToSurface.jl

```diff
@@ -0,0 +1,4 @@
+function distance_to_surface(point::AbstractCoordinatePoint{T}, c::AbstractGeometry{T}) where {T}
+    distance_list=[distance_to_surface(point,surf) for surf in surfaces(c)]
+    minimum(filter(!isnan,distance_list))
+end
```

Comment:
I guess this is a method for `AbstractVolumePrimitive` or `AbstractConstructiveGeometry` (because only for those, `surfaces` is defined). Also, this fails if all entries in `distance_list` are `NaN` (can this ever happen?)

This can probably be written in a more optimized way, something like:
```suggestion
function distance_to_surface(point::AbstractCoordinatePoint{T}, c::Union{<:AbstractConstructiveGeometry{T}, <:AbstractVolumePrimitive{T}})::T where {T}
    min_distance::T = T(Inf)
    for surf in surfaces(c)
        d::T = distance_to_surface(point, surf)
        min_distance = ifelse(isnan(d), min_distance, min(d, min_distance))
    end
    return min_distance
end
```
This would return `Inf` in case that all distances to all surfaces are `NaN`. Is that a good default value?

Reply:
Yes, only for `AbstractVolumePrimitive` or `AbstractConstructiveGeometry`.

I noticed that the `NaN` corresponds to the zero area surface (basically 2d lines). 

I haven't dug into the detail. I basically adopt the code from the pull request 138, solve the error and test on the `TrueCoaxial` and `InvertedCoax`.

We could use `Inf`, but if the final return is `Inf`, we should at least throw an warning